astroquery/sdss/core.py

```diff
@@ -94,76 +103,86 @@ def query_crossid_async(self, coordinates, obj_names=None,
         else:
             try:
-                float(radius)
-            except TypeError:
+                radius = Angle(radius).to(u.arcmin).value
+            except ValueError:
                 raise TypeError("radius should be either Quantity or "
                                 "convertible to float.")
+        if radius > self.MAX_CROSSID_RADIUS.value:
+            raise ValueError(f"radius must be less than {self.MAX_CROSSID_RADIUS}.")
 
-        sql_query = 'SELECT\r\n'  # Older versions expect the CRLF to be there.
-
-        if specobj_fields is None:
-            if photoobj_fields is None:
-                photoobj_fields = crossid_defs
-            photobj_fields = ['p.{0}'.format(i) for i in photoobj_fields]
-            photobj_fields.append('p.objID as obj_id')
-            specobj_fields = []
+        if (not isinstance(coordinates, list) and
+            not isinstance(coordinates, Column) and
+            not (isinstance(coordinates, commons.CoordClasses) and
+                 not coordinates.isscalar)):
+            coordinates = [coordinates]
+        if obj_names is None:
+            obj_names = ['obj_{0}'.format(i) for i in range(len(coordinates))]
+        elif len(obj_names) != len(coordinates):
+            raise ValueError("Number of coordinates and obj_names should "
+                         "be equal")
+        if region:
+            data = "ra dec \n"
+            data_format = '{ra} {dec}'
         else:
-            specobj_fields = ['s.{0}'.format(i) for i in specobj_fields]
-            if photoobj_fields is not None:
-                photobj_fields = ['p.{0}'.format(i) for i in photoobj_fields]
-                photobj_fields.append('p.objID as obj_id')
-            else:
-                photobj_fields = []
-                specobj_fields.append('s.SpecObjID as obj_id')
-
-        sql_query += ', '.join(photobj_fields + specobj_fields)
-
-        sql_query += ',dbo.fPhotoTypeN(p.type) as type \
-                      FROM #upload u JOIN #x x ON x.up_id = u.up_id \
-                      JOIN PhotoObjAll p ON p.objID = x.objID '
-        if specobj_fields:
-            sql_query += 'JOIN SpecObjAll s ON p.objID = s.bestObjID '
-        sql_query += 'ORDER BY x.up_id'
-
-        data = "obj_id ra dec \n"
-        data += " \n ".join(['{0} {1} {2}'.format(obj_names[i],
-                                                  coordinates[i].ra.deg,
-                                                  coordinates[i].dec.deg)
+            data = "name ra dec \n"  # SDSS's own examples default to 'name'.  'obj_id' is too easy to confuse with 'objID'
+            data_format = '{obj} {ra} {dec}'
+        data += " \n ".join([data_format.format(obj=obj_names[i],
+                                                ra=coordinates[i].ra.deg,
+                                                dec=coordinates[i].dec.deg)
                              for i in range(len(coordinates))])
```

Comment:
The `list` created here is discarded after iterating over it once, so it would be better to use a generator expression here. Furthermore, using `zip` is preferable over accessing elements by index.
```suggestion
        data += " \n ".join(data_format.format(obj=obj,
                                               ra=coord.ra.deg,
                                               dec=coord.dec.deg)
                             for obj, coord in zip(obj_names, coordinates))
```

Reply:
I was trying to make a minimal change.  Again, is there an explicit policy that replacing explicit lists with implicit iterables (which are less obvious) should be undertaken wherever possible?

Reply:
Pull requests are not required to update unrelated code, but these lines are being touched anyways.

Replacing list comprehensions with generator expressions is faster and uses less memory, but the values can be accessed only once. In this case the values are accessed only once anyways, so there really are no downsides to using a generator expression here.

Combining iterables with `zip` is faster than looping over indices. Using `IPython`:
```python
>>> word = "foo"
>>> letters = list(word)
>>> %timeit for i in range(len(word)): assert word[i] == letters[i]
392 ns ± 32.4 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
>>> %timeit for x, y in zip(word, letters): assert x == y
280 ns ± 2.81 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

By using an f-string the code snippet could be made even shorter:
```suggestion
        data += " \n ".join(f"{obj} {coord.ra.deg} {coord.dec.deg}" for obj, coord in zip(obj_names, coordinates))
```